internal/config/crd_config.go

```diff
@@ -87,75 +87,36 @@ func PopulateConfigs(configMap corev1.ConfigMap) (*MeshsyncConfig, error) {
-	globalPipelines := make(PipelineConfigs, 0)
-	localPipelines := make(PipelineConfigs, 0)
-
 	if len(meshsyncConfig.WhiteList) != 0 {
-		for _, v := range Pipelines[GlobalResourceKey] {
+		for k, v := range Pipelines[GlobalResourceKey] {
 			if idx := slices.IndexFunc(meshsyncConfig.WhiteList, func(c ResourceConfig) bool { return c.Resource == v.Name }); idx != -1 {
 				config := meshsyncConfig.WhiteList[idx]
-				v.Events = config.Events
-				globalPipelines = append(globalPipelines, v)
+				Pipelines[GlobalResourceKey][k].Events = config.Events
 			}
 		}
-		if len(globalPipelines) > 0 {
-			meshsyncConfig.Pipelines = map[string]PipelineConfigs{}
-			meshsyncConfig.Pipelines[GlobalResourceKey] = globalPipelines
-		}
-
 		// Handle local resources
-		for _, v := range Pipelines[LocalResourceKey] {
+		for k, v := range Pipelines[LocalResourceKey] {
 			if idx := slices.IndexFunc(meshsyncConfig.WhiteList, func(c ResourceConfig) bool { return c.Resource == v.Name }); idx != -1 {
 				config := meshsyncConfig.WhiteList[idx]
-				v.Events = config.Events
-				localPipelines = append(localPipelines, v)
+				Pipelines[LocalResourceKey][k].Events = config.Events
 			}
 		}
-
-		if len(localPipelines) > 0 {
-			if meshsyncConfig.Pipelines == nil {
-				meshsyncConfig.Pipelines = make(map[string]PipelineConfigs)
-			}
-			meshsyncConfig.Pipelines[LocalResourceKey] = localPipelines
-		}
-
-	} else {
+	}
+	if len(meshsyncConfig.BlackList) != 0 {
 
 		for _, v := range Pipelines[GlobalResourceKey] {
-			if idx := slices.IndexFunc(meshsyncConfig.BlackList, func(c string) bool { return c == v.Name }); idx == -1 {
-				v.Events = DefaultEvents
-				globalPipelines = append(globalPipelines, v)
+			if idx := slices.IndexFunc(meshsyncConfig.BlackList, func(c string) bool { return c == v.Name }); idx != -1 {
+				confIdx := slices.IndexFunc(Pipelines[GlobalResourceKey], func(c PipelineConfig) bool { return c.Name == v.Name })
+				Pipelines[GlobalResourceKey] = slices.Delete(Pipelines[GlobalResourceKey], confIdx, confIdx+1)
 			}
 		}
-		if len(globalPipelines) > 0 {
-			meshsyncConfig.Pipelines = map[string]PipelineConfigs{}
-			meshsyncConfig.Pipelines[GlobalResourceKey] = globalPipelines
-		}
 
 		// Handle local resources
 		for _, v := range Pipelines[LocalResourceKey] {
-			if idx := slices.IndexFunc(meshsyncConfig.BlackList, func(c string) bool { return c == v.Name }); idx == -1 {
-				v.Events = DefaultEvents
-				localPipelines = append(localPipelines, v)
-			}
-		}
-
-		if len(localPipelines) > 0 {
-			if meshsyncConfig.Pipelines == nil {
-				meshsyncConfig.Pipelines = make(map[string]PipelineConfigs)
+			if idx := slices.IndexFunc(meshsyncConfig.BlackList, func(c string) bool { return c == v.Name }); idx != -1 {
+				confIdx := slices.IndexFunc(Pipelines[LocalResourceKey], func(c PipelineConfig) bool { return c.Name == v.Name })
+				Pipelines[LocalResourceKey] = slices.Delete(Pipelines[LocalResourceKey], confIdx, confIdx+1)
```

Comment:
@KiptoonKipkurui instead of using `IndexFunc` to get the idx in `Pipelines[LocalResourceKey]`, can't we directly fetch the id from the loop's iterator? Something like;
`for confIdx, v := range Pipelines[LocalResourceKey]`